test/integration/test_session.py

```diff
@@ -71,3 +71,33 @@ def test_estimator_sampler(self, service):
             self.assertEqual(result.metadata[0]["shots"], 400)
             self.assertAlmostEqual(result.quasi_dists[0][3], 0.5, delta=0.1)
             self.assertAlmostEqual(result.quasi_dists[0][0], 0.5, delta=0.1)
+
+    @run_integration_test
+    def test_choosing_correct_backend(self, service):
+        """Test that the backend selected in options is used."""
+
+        backend = "ibmq_qasm_simulator"
+        inputs = {"iterations": 1}
+        options = {
+            "backend": backend,
+        }
+        with Session(service=service, backend="statevector_simulator") as session:
+            job = session.run(program_id="hello-world", options=options, inputs=inputs)
+            self.assertEqual(backend, job.backend().name)
+
```

Comment:
@jyu00 my understanding is that in a scenario like this the backend passed in through options should be honored?

Reply:
Yes, since the docstring says `options` can be anything in `RuntimeOptions` (which includes `backend`). We should change that though since you can't have a session cross multiple backends.